fixtures/coverall/src/lib.rs

```diff
@@ -122,11 +131,70 @@ impl Coveralls {
     fn panic(&self, message: String) {
         panic!("{}", message);
     }
+
+    fn take_other(&self, other: Option<Arc<Self>>) {
+        *self.other.lock().unwrap() = match other {
+            None => None,
+            Some(arc) => Some(Arc::clone(&arc)),
```

Comment:
IIUC, we already own the `Arc<Self>` in `other` at this point, so I don't think we need to clone here, and could just `*self.other.lock().unwrap() = other`. The current setup will increment the refcount for the `Arc::clone`, then immediately decrement it when `other` is dropped at the end of the function.